e2e/test/scenarios/visualizations-tabular/reproductions/18976-pivot-table-columns.cy.spec.js

```diff
@@ -30,4 +33,30 @@ describe("issue 18976", () => {
     // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
     cy.findByText("Showing 1 row");
   });
+
+  it("should not keep orphan columns rendered after switching from pivot to regular table (metabase#18817)", () => {
+    cy.createQuestion(
+      {
+        query: {
+          "source-table": PEOPLE_ID,
+          aggregation: [["count"]],
+          breakout: [
+            ["field", PEOPLE.NAME],
+            ["field", PEOPLE.SOURCE],
+          ],
+          limit: 5,
+        },
+        database: SAMPLE_DB_ID,
+        display: "table",
+      },
+      { visitQuestion: true },
+    );
+
+    cy.findByTestId("qb-header").button("Summarize").click();
+    cy.findByLabelText("Source")
```

Comment:
It doesn't hurt to scope this to the right sidebar, just in case we end up with multiple elements in the UI in the future.
`rightsidebar().findByLabelText(...)` should work.